### PR TITLE
merge sharing screens

### PR DESCRIPTION
**Pull request by kevodwyer**
web-ui PR: https://github.com/Peergos/web-ui/pull/156
- getLatest() does not work when files are shared
- share*AccessWith and sharedWith now take a string path. This ensures the latest version of file is retrieved.

### CODE REVIEW

File: src/peergos/gwt/emu/java/nio/file/Path.java [hunk 1]
```diff
@@ -35,7 +35,16 @@ public Path getParent() {
     }
 
     public Path getFileName() {
-        throw new IllegalArgumentException("Not implemented!");
+        int index = pathString.lastIndexOf(SEPARATOR);
+        if (index == -1) {
+            return new Path(pathString);
+        } else {
+            if(index == pathString.length() - 1) {
+                return null;
```
Comment (ianopolous) [inline src/peergos/gwt/emu/java/nio/file/Path.java:43]: why null?

File: src/peergos/shared/user/fs/FileWrapper.java [hunk 2]
```diff
@@ -109,13 +109,6 @@ public FileWrapper withVersion(Snapshot version) {
         return new FileWrapper(capTrie, pointer, linkPointer, entryWriter, ownername, version);
     }
 
-    @JsMethod
-    public CompletableFuture<FileWrapper> getLatest(NetworkAccess network) {
```
Comment (ianopolous) [inline src/peergos/shared/user/fs/FileWrapper.java:113]: aren't we going to need this in other features?

File: src/peergos/gwt/emu/java/nio/file/Path.java (hunk 1 above)
Comment (kevodwyer) [reply src/peergos/gwt/emu/java/nio/file/Path.java:43]: That is what the Path method does also.

File: src/peergos/shared/user/fs/FileWrapper.java (hunk 2 above)
Comment (kevodwyer) [reply src/peergos/shared/user/fs/FileWrapper.java:113]: For video capture - yes. But as it doesn't seem to work when the file is subsequently shared it seemed prudent to remove

File: src/peergos/gwt/emu/java/nio/file/Path.java (hunk 1 above)
Comment (ianopolous) [reply src/peergos/gwt/emu/java/nio/file/Path.java:43]: It only returns null if the path it empty: https://docs.oracle.com/javase/7/docs/api/java/nio/file/Path.html#getFileName()

We are different on e.g. /somedir/

File: src/peergos/shared/user/UserContext.java [hunk 3]
```diff
@@ -1374,15 +1374,19 @@ private static void ensureAllowedToShare(FileWrapper file, String ourname, boole
         });
     }
 
-    public CompletableFuture<Boolean> shareWriteAccessWithAll(Path fileToShare,
-                                                               Set<String> writersToAdd) {
+
+    @JsMethod
+    public CompletableFuture<Boolean> shareWriteAccessWith(String pathToFile,
```
Comment (ianopolous) [inline src/peergos/shared/user/UserContext.java:1379]: In general a typed api is better than a string api. So I'd prefer if the JS method called the typed api. I'm hoping at somepoint we can remove all these JS only apis into a sub class or something. Java code for sure should be using the typed api.

**Pull request closed without merging**